pybliometrics/sciencedirect/sciencedirect_search.py

```diff
@@ -0,0 +1,157 @@
+            new = doc(authors=authors,
+                first_author=item.get('dc:creator'),
+                doi=doi,
+                title=item.get("dc:title"),
+                link=links["scidir"],
+                load_date=item.get("load-date"),
+                openaccess_status=item.get("openaccess"),
+                pii=item.get("pii"),
+                coverDate=item.get("prism:coverDate"),
+                endingPage=item.get("prism:endingPage"),
+                publicationName=item.get("prism:publicationName"),
+                startingPage=item.get("prism:startingPage"),
+                api_link=links["api_link"] or item.get("prism:url"),
+                volume=item.get("prism:volume"))
+            out.append(new)
+        check_integrity(out, self._integrity, self._action)
+        return out or None
+
+    def __init__(self,
+                 query: str,
+                 refresh: Union[bool, int] = False,
+                 view: str = None,
+                 verbose: bool = False,
+                 download: bool = True,
+                 integrity_fields: Union[List[str], Tuple[str, ...]] = None,
+                 integrity_action: str = "raise",
+                 subscriber: bool = True,
+                 **kwds: str
+                 ) -> None:
+        """Interaction with the ScienceDirect Search API. This represents a search against the
+        ScienceDirect cluster, which contains serial/nonserial full-text articles. Note that this API
+        replicates the search experience on `Science Direct <www.sciencedirect.com>`__.
+
+        :param query: A string of the query as used in the `Science Direct Search <https://dev.elsevier.com/tecdoc_sdsearch_migration.html>`__. Note that
+        this library uses the GET interface.
+        :param refresh: Whether to refresh the cached file if it exists or not.
+                        If int is passed, cached file will be refreshed if the
+                        number of days since last modification exceeds that value.
+        :param view: Which view to use for the query, see `the documentation <https://dev.elsevier.com/sd_search_views.html>`__.
+                     Allowed values: `STANDARD`.
+        :param verbose: Whether to print a download progress bar.
+        :param download: Whether to download results (if they have not been
+                         cached).
+        :param integrity_fields: A list or tuple with the names of fields whose completeness should
+                                 be checked.  `ArticleMetadata` will perform the
+                                 action specified in `integrity_action` if
+                                 elements in these fields are missing.  This
+                                 helps avoiding idiosynchratically missing
+                                 elements that should always be present
+                                 (e.g., EID or source ID).
```

Comment:
Since neither EID nor source ID can be present in the output, we need different examples.